capsules/src/analog_comparator.rs

```diff
@@ -117,7 +124,19 @@ impl<'a, A: hil::analog_comparator::AnalogComparator<'a>> Driver for AnalogCompa
     /// - `3`: Stop interrupt-based comparisons.
     ///        Input x chooses the desired comparator ACx (e.g. 0 or 1 for
     ///        hail, 0-3 for imix)
-    fn command(&self, command_num: usize, channel: usize, _: usize, _: AppId) -> CommandReturn {
+    fn command(&self, command_num: usize, channel: usize, _: usize, appid: AppId) -> CommandReturn {
+        // Check if this driver is free, or already dedicated to this process.
+        let match_or_empty_or_nonexistant = self.current_process.map_or(true, |current_process| {
+            self.grants
+                .enter(*current_process, |_, _| current_process == &appid)
+                .unwrap_or(true)
+        });
+        if match_or_empty_or_nonexistant {
+            self.current_process.set(appid);
+        } else {
+            return CommandReturn::failure(ErrorCode::NOMEM);
+        }
+
         match command_num {
```

Comment:
One concern with this approach: it makes it possible for the vale returned by the "driver exists" check (aka command 0) to change over time, if another process claims the driver but later exits.

I guess this is just part of dealing with the existence of non-virtualized drivers. Anyone running multiple apps trying to use one of these capsules is simply doing something wrong.